tests/next_tests/unit_tests/type_system_tests/test_type_translation.py

```diff
@@ -199,3 +200,40 @@ def test_from_value_module():
     assert type_translation.from_value(dummy_package.dummy_module.dummy_int) == ts.ScalarType(
         kind=ts.ScalarKind.INT32
     )
+
+
+class SomeEnum(eve.IntEnum):
+    FOO = 1
+
+
+@pytest.mark.parametrize(
+    "value, type_, expected",
+    [
+        (gtx.int32(1), ts.ScalarType(kind=ts.ScalarKind.INT32), gtx.int32(1)),
+        (gtx.int64(1), ts.ScalarType(kind=ts.ScalarKind.INT64), gtx.int64(1)),
+        (1.0, ts.ScalarType(kind=ts.ScalarKind.INT64), gtx.int64(1)),
```

Comment:
Is this really intended behavior? I don't understand why... What about `1.1`? It looks to me like this should trigger an error...
```suggestion
        (1.0, ts.ScalarType(kind=ts.ScalarKind.INT64), gtx.int64(1)),
```

Reply:
This is the intended behavior as explained in the function docstring. To avoid misunderstandings I propose to rename the function to `unsafe_cast_to`.